IanGRap/Squish
# Dash bounce

Contributors: Austin

Goal:
- Add a bounce back when the player collides with objects during a dash

What Changed:
- I added collision events to PlayerDash, which adds a small force when the player collides during a dash
- I added an option to PlayerMovement which disables movement while the player is bouncing

Customization:
- PlayerDash
    - BounceOnDashCollision (enables / disables the feature)
    - BounceBackForce (the force modifier of the bounce)
    - BounceBackUpwardTravel (how far upwards the player travels in a bounce)
    - BounceResetTimer (the amount of time before the player is considered to be not bouncing)
    - BounceBackLayerMask (controls which objects the player can bounce off of)
- PlayerMovement
    - MoveWhileBouncing (disables player movement while bouncing [this is where the BounceResetTimer is useful])

How to Test:
- Load up both the lobby and the match scene and make sure you bounce off of the objects in the LayerMask (by default its everything but the floor)
- Test all options that I laid out in Customization
